include/picongpu/particles/collision/detail/ListEntry.hpp

```diff
@@ -64,8 +64,18 @@ namespace picongpu
                             for(int numTries = 0; numTries < maxTries; ++numTries)
                             {
 #if(BOOST_LANG_CUDA || BOOST_COMP_HIP) // Allocate memory on a GPU device
-                                ptrToIndicies
-                                    = (uint32_t*) deviceHeapHandle.malloc(worker.getAcc(), sizeof(uint32_t) * numPar);
+                                constexpr uint32_t typicalParticlesPerSupercell
+                                    = particles::TYPICAL_PARTICLES_PER_CELL;
+                                /* Round-up the number of slots up to an mutiple of the typical amount of particles of
+                                 * a cell. This will waist memory but is avoiding that the mallocMC heap is fragmented
+                                 * which often results into out of memory crashes even if the amount of particles in
+                                 * the simulation is small.
+                                 */
```

Comment:
Maybe add a static assert that `typicalParticlesPerSupercell` is not 0, it shouldn't be, but it is sth that is usually set by the user in a param file.

Reply:
done